eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentLifecycle.scala

```diff
@@ -219,9 +214,23 @@ object PaymentLifecycle {
   case class UnreadableRemoteFailure(route: Seq[Hop]) extends PaymentFailure
   case class PaymentFailed(id: UUID, paymentHash: ByteVector32, failures: Seq[PaymentFailure]) extends PaymentResult
 
-  sealed trait PaymentOptions
-  case object LegacyPayload extends PaymentOptions
-  case object TlvPayload extends PaymentOptions
+  /**
+   * Options to help build the final payload of the payment route.
+   */
+  sealed trait PaymentOptions {
+    // The final htlc amount in millisatoshis.
+    val finalAmountMsat: Long
+    // The final htlc expiry in number of blocks.
+    val finalExpiry: Long
+  }
+  case class LegacyPayload(finalAmountMsat: Long, finalCltvExpiry: Long) extends PaymentOptions {
```

Comment:
`finalCltvExpiry` should be `finalCltvExpiryDelta`

Reply:
Yeah I hesitated a lot with that one :)